apps/aeutil/lib/serialization.ex

```diff
@@ -97,26 +75,99 @@ defmodule Aeutil.Serialization do
     if is_tuple(head) do
       merkle_proof(Tuple.to_list(head), acc)
     else
-      acc = [bech32_binary(head, :account, :serialize) | acc]
+      acc = [serialize_value(head, :account) | acc]
       merkle_proof(tail, acc)
     end
   end
 
   @spec pack_binary(term()) :: map()
```

Comment:
This looks a bit confusing, I think it's better to keep it the way it was before (have a case depending on the data which can be extended), it seems hard to handle each possible case in advance and have it done with code that looks good. Also, the map and list packs look wrong, the list one returns a list in the end when it should be a msgpack binary and the map one packs each value and then packs the whole thing again which gives an incorrect result.

Reply:
I'm looking into this